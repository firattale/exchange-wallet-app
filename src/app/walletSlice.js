import { createSlice } from '@reduxjs/toolkit';

export const walletSlice = createSlice({
  name: 'wallet',
  initialState: {
    USD: 40.00,
    GBP: 50.00,
    EUR: 60.00,
    selectedAmount: "",
  },
  reducers: {
    changeSelectedAmount: (state, action) => {
      const { selectedAmount } = action.payload
      state.selectedAmount = selectedAmount;
    },
    updateWallet: (state, action) => {
      const { firstWalletType, firstAmount, secondWalletType, secondAmount } = action.payload
      state[firstWalletType] = firstAmount;
      state[secondWalletType] = secondAmount;
      state.selectedAmount = ""
    },

  },
});

export const { changeSelectedAmount, updateWallet } = walletSlice.actions;

export const selectedAmount = state => state.wallet.selectedAmount;
export const selectUsdWallet = state => state.wallet.usdWallet;

export default walletSlice.reducer;
