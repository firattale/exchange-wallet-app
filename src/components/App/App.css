body,
html,
#root {
    height: 100%;
    width: 100%;
    max-width: 600px;
    margin: auto;
}