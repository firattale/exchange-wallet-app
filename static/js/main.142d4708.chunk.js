(this.webpackJsonprevolut=this.webpackJsonprevolut||[]).push([[0],{17:function(e,t,n){},19:function(e,t,n){e.exports=n(33)},24:function(e,t,n){},25:function(e,t,n){},26:function(e,t,n){},27:function(e,t,n){},33:function(e,t,n){"use strict";n.r(t);var c=n(0),r=n.n(c),a=n(9),o=n.n(a),l=(n(24),n(25),n(26),n(27),n(2)),u=n(7),i=Object(u.b)({name:"exchange",initialState:{firstCurrency:"GBP",firstSign:"\xa3",secondCurrency:"USD",currencyRate:"",secondSign:"$"},reducers:{changeFirstCurrency:function(e,t){var n=t.payload,c=n.first,r=n.firstSign;e.firstCurrency=c,e.firstSign=r},changeSecondCurrency:function(e,t){var n=t.payload,c=n.second,r=n.secondSign;e.secondCurrency=c,e.secondSign=r},changeCurrencyRate:function(e,t){e.currencyRate=t.payload}}}),s=i.actions,m=s.changeFirstCurrency,f=s.changeSecondCurrency,d=s.changeCurrencyRate,y=function(e){return e.exchange.firstCurrency},g=function(e){return e.exchange.secondCurrency},b=function(e){return e.exchange.currencyRate},v=function(e){return e.exchange.firstSign},E=function(e){return e.exchange.secondSign},h=i.reducer,p=Object(u.b)({name:"wallet",initialState:{USD:40,GBP:50,EUR:60,firstAmount:"",secondAmount:"",firstWalletError:null},reducers:{changeFirstAmount:function(e,t){var n=t.payload.firstAmount;e.firstAmount=n},changeSecondAmount:function(e,t){var n=t.payload.secondAmount;e.secondAmount=n},checkFirstWalletError:function(e,t){var n=t.payload.error;e.firstWalletError=n},updateWallet:function(e,t){var n=t.payload,c=n.firstWalletType,r=n.firstAmount,a=n.secondWalletType,o=n.secondAmount;e[c]=r,e[a]=o,e.firstAmount="",e.secondAmount=""}}}),w=p.actions,j=w.changeFirstAmount,A=w.changeSecondAmount,N=w.updateWallet,x=w.checkFirstWalletError,C=function(e){return e.wallet.firstAmount},O=function(e){return e.wallet.secondAmount},S=function(e){return e.wallet.firstWalletError},F=p.reducer,R={USD:"$",GBP:"\xa3",EUR:"\u20ac"},W=function(e,t){var n=e.first,c=e.second,r="".concat("https://api.exchangeratesapi.io","/latest?base=").concat(n,"&symbols=").concat(c);return fetch(r).then((function(e){return e.json()})).then((function(e){if(e.error)throw new Error("There is something wrong with your request. Please try again!");t(d(e.rates[c].toFixed(4)))})).catch((function(e){alert(e)}))},k=function(){for(var e=setTimeout(";"),t=0;t<e;t++)clearTimeout(t)},U=function(e,t){if(Number.isInteger(Number(e)))t(x({error:null}));else{var n=e.split(".")[1].length||0;t(x(n<3?{error:null}:{error:"Your input must have only two digits after comma"}))}},B=function(e){return function(t){k(),W(e,t),setInterval((function(){W(e,t)}),1e4)}},T=function(){var e=Object(l.c)(y),t=Object(l.c)(g),n=Object(l.c)(b),a=Object(l.c)(v),o=Object(l.c)(E),u=Object(l.b)();return Object(c.useEffect)((function(){u(B({first:e,second:t}))}),[e,t]),r.a.createElement("div",{className:"exchange-container"},r.a.createElement("span",{className:"currency"},a),"1=",r.a.createElement("span",{className:"currency"},o),n.slice(0,4),r.a.createElement("span",{className:"currency"},n.slice(4,6)))},P=n(34),D=function(e){var t=e.firstAmount,n=e.firstCurrency,c=e.secondCurrency,a=e.currencyRate,o=Object(l.c)((function(e){return e.wallet[n].toFixed(2)})),u=Object(l.c)((function(e){return e.wallet[c].toFixed(2)})),i=Object(l.c)(S),s=Boolean(+t>o||""===t||n===c||i),m=Object(l.b)();return r.a.createElement("div",{className:"header-container"},r.a.createElement(P.a,{type:"button",className:"header-button"},"Cancel"),r.a.createElement(T,null),r.a.createElement(P.a,{type:"button",className:"header-button",disabled:s,onClick:function(){m(N({firstWalletType:n,secondWalletType:c,firstAmount:+o-t,secondAmount:+u+t*a}))}},"Exchange"))},G=(n(17),n(35)),I=function(e){var t=e.currency,n=e.sign,c=e.secondCurrency,a=e.currencyRate,o=e.firstAmount,u=Object(l.b)(),i=Object(l.c)((function(e){return e.wallet[t].toFixed(2)})),s=Object(l.c)(S);return r.a.createElement(r.a.Fragment,null,r.a.createElement("div",{className:"wallet-container background"},r.a.createElement("div",{className:"d-flex flex-column justify-content-between h-100"},r.a.createElement("div",{className:"d-flex justify-content-between align-items-center"},r.a.createElement(G.a,{type:"select",name:"select",className:"wallet-currency background",defaultValue:t,onChange:function(e){return t=e.target.value,u(m({first:t,firstSign:R[t]})),u(j({firstAmount:""})),void u(A({secondAmount:""}));var t}},r.a.createElement("option",null,"GBP"),r.a.createElement("option",null,"USD"),r.a.createElement("option",null,"EUR")),r.a.createElement(G.a,{type:"number",className:"wallet-input",value:o,onChange:function(e){return t=e.target.value,U(t,u),u(j({firstAmount:t})),void u(A({secondAmount:(t*a).toFixed(2)}));var t},disabled:t===c,min:"0"})),s&&r.a.createElement("p",{className:"wallet-error-message"},s),r.a.createElement("div",null,r.a.createElement("div",{className:"wallet-pocket"},"You Have ",n,i)))),r.a.createElement("div",{className:"arrow-down"}))},Y=function(e){var t=e.firstSign,n=e.firstCurrency,c=e.currency,a=e.secondSign,o=e.currencyRate,u=Object(l.b)(),i=Object(l.c)(O),s=(1/o).toFixed(2),m=Object(l.c)((function(e){return e.wallet[c].toFixed(2)}));return r.a.createElement("div",{className:"wallet-container"},r.a.createElement("div",{className:"d-flex flex-column justify-content-between h-100"},r.a.createElement("div",{className:"d-flex justify-content-between align-items-center"},r.a.createElement(G.a,{type:"select",name:"select",className:"wallet-currency-second",defaultValue:c,onChange:function(e){return t=e.target.value,u(f({second:t,secondSign:R[t]})),u(j({firstAmount:""})),void u(A({secondAmount:""}));var t}},r.a.createElement("option",null,"GBP"),r.a.createElement("option",null,"USD"),r.a.createElement("option",null,"EUR")),r.a.createElement(G.a,{type:"number",className:"wallet-input-second",value:i,onChange:function(e){return function(e){var t=e*(1/o);U(e,u),u(j({firstAmount:t.toFixed(2)})),u(A({secondAmount:e}))}(e.target.value)},disabled:c===n,min:"0"})),r.a.createElement("div",{className:"d-flex justify-content-between"},r.a.createElement("div",{className:"wallet-pocket"},"You Have ",a,m),r.a.createElement("div",{className:"wallet-pocket"},"1",a,"=",t,s))))},$=function(){var e=Object(l.c)(y),t=Object(l.c)(g),n=Object(l.c)(v),c=Object(l.c)(E),a=Object(l.c)(C),o=Object(l.c)(b);return r.a.createElement("div",{className:"h-100 w-100"},r.a.createElement(D,{firstAmount:a,firstCurrency:e,secondCurrency:t,currencyRate:o}),r.a.createElement("div",{className:"wallets-container"},r.a.createElement(I,{currency:e,firstAmount:a,secondCurrency:t,sign:n,currencyRate:o}),r.a.createElement(Y,{currency:t,firstCurrency:e,firstSign:n,secondSign:c,currencyRate:o})))},H=Object(u.a)({reducer:{exchange:h,wallet:F}});Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));o.a.render(r.a.createElement(r.a.StrictMode,null,r.a.createElement(l.a,{store:H},r.a.createElement($,null))),document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then((function(e){e.unregister()}))}},[[19,1,2]]]);
//# sourceMappingURL=main.142d4708.chunk.js.map