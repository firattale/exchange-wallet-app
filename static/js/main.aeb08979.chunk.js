(this.webpackJsonpcurrency_exchange_wallet=this.webpackJsonpcurrency_exchange_wallet||[]).push([[0],{17:function(e,t,n){},19:function(e,t,n){e.exports=n(33)},24:function(e,t,n){},25:function(e,t,n){},26:function(e,t,n){},27:function(e,t,n){},33:function(e,t,n){"use strict";n.r(t);var c=n(0),a=n.n(c),r=n(9),l=n.n(r),o=(n(24),n(25),n(26),n(27),n(2)),u=n(7),i=Object(u.b)({name:"exchange",initialState:{firstCurrency:"GBP",firstSign:"\xa3",secondCurrency:"USD",currencyRate:"",secondSign:"$"},reducers:{changeFirstCurrency:function(e,t){var n=t.payload,c=n.first,a=n.firstSign;e.firstCurrency=c,e.firstSign=a},changeSecondCurrency:function(e,t){var n=t.payload,c=n.second,a=n.secondSign;e.secondCurrency=c,e.secondSign=a},changeCurrencyRate:function(e,t){e.currencyRate=t.payload}}}),s=i.actions,m=s.changeFirstCurrency,d=s.changeSecondCurrency,f=s.changeCurrencyRate,y=function(e){return e.exchange.firstCurrency},g=function(e){return e.exchange.secondCurrency},b=function(e){return e.exchange.currencyRate},E=function(e){return e.exchange.firstSign},p=function(e){return e.exchange.secondSign},v=i.reducer,h=Object(u.b)({name:"wallet",initialState:{USD:40,GBP:50,EUR:60,firstAmount:"",secondAmount:"",firstWalletError:null},reducers:{changeFirstAmount:function(e,t){var n=t.payload.firstAmount;e.firstAmount=n},changeSecondAmount:function(e,t){var n=t.payload.secondAmount;e.secondAmount=n},checkFirstWalletError:function(e,t){var n=t.payload.error;e.firstWalletError=n},updateWallet:function(e,t){var n=t.payload,c=n.firstWalletType,a=n.firstAmount,r=n.secondWalletType,l=n.secondAmount;e[c]=a,e[r]=l,e.firstAmount="",e.secondAmount=""}}}),w=h.actions,j=w.changeFirstAmount,A=w.changeSecondAmount,x=w.updateWallet,N=w.checkFirstWalletError,C=function(e){return e.wallet.firstAmount},O=function(e){return e.wallet.secondAmount},S=function(e){return e.wallet.firstWalletError},F=h.reducer,R={USD:"$",GBP:"\xa3",EUR:"\u20ac"},W=function(e,t){var n=e.first,c=e.second,a="".concat("https://api.exchangeratesapi.io","/latest?base=").concat(n,"&symbols=").concat(c);return fetch(a).then((function(e){return e.json()})).then((function(e){if(e.error)throw new Error("There is something wrong with your request. Please try again!");t(f(e.rates[c].toFixed(4)))})).catch((function(e){alert(e)}))},k=function(){for(var e=setTimeout(";"),t=0;t<e;t++)clearTimeout(t)},U=function(e,t){var n=Number(e);if(!Number.isInteger(n)&&n>0){var c=e.split(".")[1].length||0;t(N(c<3?{error:null}:{error:"Your input must have only two digits after comma"}))}else t(N(n<0?{error:"Your input must be a positive number"}:{error:null}))},B=function(e){return function(t){k(),W(e,t),setInterval((function(){W(e,t)}),1e4)}},T=function(){var e=Object(o.c)(y),t=Object(o.c)(g),n=Object(o.c)(b),r=Object(o.c)(E),l=Object(o.c)(p),u=Object(o.b)();return Object(c.useEffect)((function(){u(B({first:e,second:t}))}),[e,t]),a.a.createElement("div",{className:"exchange-container"},a.a.createElement("span",{className:"currency"},r),"1=",a.a.createElement("span",{className:"currency"},l),n.slice(0,4),a.a.createElement("span",{className:"currency"},n.slice(4,6)))},P=n(34),D=function(e){var t=e.firstAmount,n=e.firstCurrency,c=e.secondCurrency,r=e.currencyRate,l=Object(o.c)((function(e){return e.wallet[n].toFixed(2)})),u=Object(o.c)((function(e){return e.wallet[c].toFixed(2)})),i=Object(o.c)(S),s=Boolean(+t>l||""===t||n===c||i),m=Object(o.b)();return a.a.createElement("div",{className:"header-container"},a.a.createElement(P.a,{type:"button",className:"header-button"},"Cancel"),a.a.createElement(T,null),a.a.createElement(P.a,{type:"button","data-testid":"exchangeButton",className:"header-button",disabled:s,onClick:function(){m(x({firstWalletType:n,secondWalletType:c,firstAmount:+l-t,secondAmount:+u+t*r}))}},"Exchange"))},G=(n(17),n(35)),Y=function(e){var t=e.currency,n=e.sign,c=e.secondCurrency,r=e.currencyRate,l=e.firstAmount,u=Object(o.b)(),i=Object(o.c)((function(e){return e.wallet[t].toFixed(2)})),s=Object(o.c)(S);return a.a.createElement(a.a.Fragment,null,a.a.createElement("div",{className:"wallet-container background"},a.a.createElement("div",{className:"d-flex flex-column justify-content-between h-100"},a.a.createElement("div",{className:"d-flex justify-content-between align-items-center"},a.a.createElement(G.a,{type:"select","data-testid":"wallet-select",name:"select",className:"wallet-currency background",defaultValue:t,onChange:function(e){return t=e.target.value,u(m({first:t,firstSign:R[t]})),u(j({firstAmount:""})),void u(A({secondAmount:""}));var t}},a.a.createElement("option",null,"GBP"),a.a.createElement("option",null,"USD"),a.a.createElement("option",null,"EUR")),a.a.createElement("input",{type:"number","data-testid":"wallet-input-1",className:"wallet-input",value:l,onChange:function(e){return t=e.target.value,U(t,u),u(j({firstAmount:t})),void u(A({secondAmount:(t*r).toFixed(2)}));var t},disabled:t===c,min:"0"})),s&&a.a.createElement("p",{className:"wallet-error-message"},s),a.a.createElement("div",null,a.a.createElement("div",{className:"wallet-pocket"},"You Have ",n,i)))),a.a.createElement("div",{className:"arrow-down"}))},_=function(e){var t=e.firstSign,n=e.firstCurrency,c=e.currency,r=e.secondSign,l=e.currencyRate,u=Object(o.b)(),i=Object(o.c)(O),s=(1/l).toFixed(2),m=Object(o.c)((function(e){return e.wallet[c].toFixed(2)}));return a.a.createElement("div",{className:"wallet-container"},a.a.createElement("div",{className:"d-flex flex-column justify-content-between h-100"},a.a.createElement("div",{className:"d-flex justify-content-between align-items-center"},a.a.createElement(G.a,{type:"select","data-testid":"wallet-select",name:"select",className:"wallet-currency-second",defaultValue:c,onChange:function(e){return t=e.target.value,u(d({second:t,secondSign:R[t]})),u(j({firstAmount:""})),void u(A({secondAmount:""}));var t}},a.a.createElement("option",null,"GBP"),a.a.createElement("option",null,"USD"),a.a.createElement("option",null,"EUR")),a.a.createElement("input",{type:"number","data-testid":"wallet-input-2",className:"wallet-input-second",value:i,onChange:function(e){return function(e){var t=e*(1/l);U(e,u),u(j({firstAmount:t.toFixed(2)})),u(A({secondAmount:e}))}(e.target.value)},disabled:c===n,min:"0"})),a.a.createElement("div",{className:"d-flex justify-content-between"},a.a.createElement("div",{className:"wallet-pocket"},"You Have ",r,m),a.a.createElement("div",{className:"wallet-pocket"},"1",r,"=",t,s))))},I=function(){var e=Object(o.c)(y),t=Object(o.c)(g),n=Object(o.c)(E),c=Object(o.c)(p),r=Object(o.c)(C),l=Object(o.c)(b);return a.a.createElement("div",{className:"h-100 w-100"},a.a.createElement(D,{firstAmount:r,firstCurrency:e,secondCurrency:t,currencyRate:l}),a.a.createElement("div",{className:"wallets-container"},a.a.createElement(Y,{currency:e,firstAmount:r,secondCurrency:t,sign:n,currencyRate:l}),a.a.createElement(_,{currency:t,firstCurrency:e,firstSign:n,secondSign:c,currencyRate:l})))},H=Object(u.a)({reducer:{exchange:v,wallet:F}});l.a.render(a.a.createElement(a.a.StrictMode,null,a.a.createElement(o.a,{store:H},a.a.createElement(I,null))),document.getElementById("root"))}},[[19,1,2]]]);
//# sourceMappingURL=main.aeb08979.chunk.js.map