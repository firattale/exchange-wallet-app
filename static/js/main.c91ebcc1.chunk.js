(this.webpackJsonprevolut=this.webpackJsonprevolut||[]).push([[0],{17:function(e,t,n){},19:function(e,t,n){e.exports=n(33)},24:function(e,t,n){},25:function(e,t,n){},26:function(e,t,n){},27:function(e,t,n){},33:function(e,t,n){"use strict";n.r(t);var c=n(0),r=n.n(c),a=n(9),o=n.n(a),l=(n(24),n(25),n(26),n(27),n(2)),i=n(7),u=Object(i.b)({name:"wallet",initialState:{USD:40,GBP:50,EUR:60,firstAmount:"",secondAmount:"",firstWalletError:null},reducers:{changeFirstAmount:function(e,t){var n=t.payload.firstAmount;e.firstAmount=n},changeSecondAmount:function(e,t){var n=t.payload.secondAmount;e.secondAmount=n},checkFirstWalletError:function(e,t){var n=t.payload.error;e.firstWalletError=n},updateWallet:function(e,t){var n=t.payload,c=n.firstWalletType,r=n.firstAmount,a=n.secondWalletType,o=n.secondAmount;e[c]=r,e[a]=o,e.firstAmount="",e.secondAmount=""}}}),s=u.actions,m=s.changeFirstAmount,d=s.changeSecondAmount,f=s.updateWallet,y=s.checkFirstWalletError,g=function(e){return e.wallet.firstAmount},b=function(e){return e.wallet.secondAmount},v=function(e){return e.wallet.firstWalletError},E=u.reducer,h=function(e,t){if(Number.isInteger(Number(e)))t(y({error:null}));else{var n=e.split(".")[1].length||0;t(y(n<3?{error:null}:{error:"Your input must have only two digits after comma"}))}},p={USD:"$",GBP:"\xa3",EUR:"\u20ac"},w=Object(i.b)({name:"exchange",initialState:{firstCurrency:"GBP",firstSign:"\xa3",secondCurrency:"USD",currencyRate:"",secondSign:"$"},reducers:{changeFirstCurrency:function(e,t){var n=t.payload,c=n.first,r=n.firstSign;e.firstCurrency=c,e.firstSign=r},changeSecondCurrency:function(e,t){var n=t.payload,c=n.second,r=n.secondSign;e.secondCurrency=c,e.secondSign=r},changeCurrencyRate:function(e,t){e.currencyRate=t.payload}}}),j=w.actions,A=j.changeFirstCurrency,N=j.changeSecondCurrency,x=j.changeCurrencyRate,C=function(e){return function(t){!function(){for(var e=setTimeout(";"),t=0;t<e;t++)clearTimeout(t)}(),function(e,t){var n=e.first,c=e.second,r="".concat("https://api.exchangeratesapi.io","/latest?base=").concat(n,"&symbols=").concat(c);fetch(r).then((function(e){return e.json()})).then((function(e){if(e.error)throw new Error("There is something wrong with your request. Please try again!");t(x(e.rates[c].toFixed(4)))})).catch((function(e){alert(e)}))}(e,t)}},O=function(e){return e.exchange.firstCurrency},S=function(e){return e.exchange.secondCurrency},F=function(e){return e.exchange.currencyRate},R=function(e){return e.exchange.firstSign},W=function(e){return e.exchange.secondSign},k=w.reducer,U=function(e){var t=Object(l.c)(O),n=Object(l.c)(S),a=Object(l.c)(F),o=Object(l.c)(R),i=Object(l.c)(W),u=Object(l.b)();return Object(c.useEffect)((function(){u(C({first:t,second:n}))}),[u,t,n]),r.a.createElement("div",{className:"exchange-container"},r.a.createElement("span",{className:"currency"},o),"1=",r.a.createElement("span",{className:"currency"},i),a.slice(0,4),r.a.createElement("span",{className:"currency"},a.slice(4,6)))},B=n(34),T=function(e){var t=e.firstAmount,n=e.firstCurrency,c=e.secondCurrency,a=e.currencyRate,o=Object(l.c)((function(e){return e.wallet[n].toFixed(2)})),i=Object(l.c)((function(e){return e.wallet[c].toFixed(2)})),u=Object(l.c)(v),s=Boolean(+t>o||""===t||n===c||u),m=Object(l.b)();return r.a.createElement("div",{className:"header-container"},r.a.createElement(B.a,{type:"button",className:"header-button"},"Cancel"),r.a.createElement(U,null),r.a.createElement(B.a,{type:"button",className:"header-button",disabled:s,onClick:function(){m(f({firstWalletType:n,secondWalletType:c,firstAmount:+o-t,secondAmount:+i+t*a}))}},"Exchange"))},P=(n(17),n(35)),D=function(e){var t=e.currency,n=e.sign,c=e.secondCurrency,a=e.currencyRate,o=e.firstAmount,i=Object(l.b)(),u=Object(l.c)((function(e){return e.wallet[t].toFixed(2)})),s=Object(l.c)(v);return r.a.createElement(r.a.Fragment,null,r.a.createElement("div",{className:"wallet-container background"},r.a.createElement("div",{className:"d-flex flex-column justify-content-between h-100"},r.a.createElement("div",{className:"d-flex justify-content-between align-items-center"},r.a.createElement(P.a,{type:"select",name:"select",className:"wallet-currency background",defaultValue:t,onChange:function(e){return t=e.target.value,i(A({first:t,firstSign:p[t]})),i(m({firstAmount:""})),void i(d({secondAmount:""}));var t}},r.a.createElement("option",null,"GBP"),r.a.createElement("option",null,"USD"),r.a.createElement("option",null,"EUR")),r.a.createElement(P.a,{type:"number",className:"wallet-input",value:o,onChange:function(e){return t=e.target.value,h(t,i),i(m({firstAmount:t})),void i(d({secondAmount:(t*a).toFixed(2)}));var t},disabled:t===c,min:"0"})),s&&r.a.createElement("p",{className:"wallet-error-message"},s),r.a.createElement("div",null,r.a.createElement("div",{className:"wallet-pocket"},"You Have ",n,u)))),r.a.createElement("div",{className:"arrow-down"}))},G=function(e){var t=e.firstSign,n=e.firstCurrency,c=e.currency,a=e.secondSign,o=e.currencyRate,i=Object(l.b)(),u=Object(l.c)(b),s=(1/o).toFixed(2),f=Object(l.c)((function(e){return e.wallet[c].toFixed(2)}));return r.a.createElement("div",{className:"wallet-container"},r.a.createElement("div",{className:"d-flex flex-column justify-content-between h-100"},r.a.createElement("div",{className:"d-flex justify-content-between align-items-center"},r.a.createElement(P.a,{type:"select",name:"select",className:"wallet-currency-second",defaultValue:c,onChange:function(e){return t=e.target.value,i(N({second:t,secondSign:p[t]})),i(m({firstAmount:""})),void i(d({secondAmount:""}));var t}},r.a.createElement("option",null,"GBP"),r.a.createElement("option",null,"USD"),r.a.createElement("option",null,"EUR")),r.a.createElement(P.a,{type:"number",className:"wallet-input-second",value:u,onChange:function(e){return function(e){var t=e*(1/o);h(e,i),i(m({firstAmount:t.toFixed(2)})),i(d({secondAmount:e}))}(e.target.value)},disabled:c===n,min:"0"})),r.a.createElement("div",{className:"d-flex justify-content-between"},r.a.createElement("div",{className:"wallet-pocket"},"You Have ",a,f),r.a.createElement("div",{className:"wallet-pocket"},"1",a,"=",t,s))))},Y=function(){var e=Object(l.c)(O),t=Object(l.c)(S),n=Object(l.c)(R),c=Object(l.c)(W),a=Object(l.c)(g),o=Object(l.c)(F);return r.a.createElement("div",{className:"h-100 w-100"},r.a.createElement(T,{firstAmount:a,firstCurrency:e,secondCurrency:t,currencyRate:o}),r.a.createElement("div",{className:"wallets-container"},r.a.createElement(D,{currency:e,firstAmount:a,secondCurrency:t,sign:n,currencyRate:o}),r.a.createElement(G,{currency:t,firstCurrency:e,firstSign:n,secondSign:c,currencyRate:o})))},$=Object(i.a)({reducer:{exchange:k,wallet:E}});Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));o.a.render(r.a.createElement(r.a.StrictMode,null,r.a.createElement(l.a,{store:$},r.a.createElement(Y,null))),document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then((function(e){e.unregister()}))}},[[19,1,2]]]);
//# sourceMappingURL=main.c91ebcc1.chunk.js.map