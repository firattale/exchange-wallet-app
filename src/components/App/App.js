import React from 'react';
import './App.css';
import Header from '../Header/Header.react'
import { useSelector } from 'react-redux';
import {
  selectFirstCurrency,
  selectSecondCurrency,
  selectFirstSign,
  selectSecondSign,
  selectCurrencyRate
} from '../../app/exchangeSlice';
import { selectFirstAmount } from '../../app/walletSlice.js';
import FirstWallet from '../Wallet/FirstWallet.react';
import SecondWallet from '../Wallet/SecondWallet.react';

export const App = () => {
  const firstCurrency = useSelector(selectFirstCurrency);
  const secondCurrency = useSelector(selectSecondCurrency);
  const firstSign = useSelector(selectFirstSign);
  const secondSign = useSelector(selectSecondSign);
  const firstAmount = useSelector(selectFirstAmount);
  const currencyRate = useSelector(selectCurrencyRate);

  return (
    <div className="h-100 w-100">
      <Header firstAmount={firstAmount} firstCurrency={firstCurrency} secondCurrency={secondCurrency} currencyRate={currencyRate} />
      <div className="wallets-container">
        <FirstWallet currency={firstCurrency} firstAmount={firstAmount} secondCurrency={secondCurrency} sign={firstSign} currencyRate={currencyRate} />
        <SecondWallet currency={secondCurrency} firstCurrency={firstCurrency} firstSign={firstSign} secondSign={secondSign} currencyRate={currencyRate} />
      </div>
    </div>
  );
}

export default App;
