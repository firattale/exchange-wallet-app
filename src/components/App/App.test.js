import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';
import { useSelector } from 'react-redux';
import {
    selectFirstCurrency,
    selectSecondCurrency,
    selectFirstSign,
    selectSecondSign,
    selectCurrencyRate
} from '../../app/exchangeSlice';
import { selectFirstAmount } from '../../app/walletSlice.js';

jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn()
}));
describe('App', () => {
    const mockAppState = {
        exchange: {
            firstCurrency: "GBP",
            firstSign: "Â£",
            secondCurrency: "USD",
            currencyRate: "",
            secondSign: "$"
        },
        wallet: {
            USD: 40.00,
            GBP: 50.00,
            EUR: 60.00,
            firstAmount: "",
            secondAmount: "",
            firstWalletError: null
        }
    }
    beforeEach(() => {
        useSelector.mockImplementation(callback => {
            return callback(mockAppState);
        });
    });
    afterEach(() => {
        useSelector.mockClear();
    });
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<App debug />);

        expect(component).toMatchSnapshot();
    });
    it('should call useSelector 6 times', () => {
        const component = shallow(<App debug />);
        expect(useSelector).toHaveBeenCalledTimes(6)
        expect(useSelector).toHaveBeenCalledWith(selectFirstCurrency)
        expect(useSelector).toHaveBeenCalledWith(selectSecondCurrency)
        expect(useSelector).toHaveBeenCalledWith(selectFirstSign)
        expect(useSelector).toHaveBeenCalledWith(selectSecondSign)
        expect(useSelector).toHaveBeenCalledWith(selectCurrencyRate)
        expect(useSelector).toHaveBeenCalledWith(selectFirstAmount)
    });
});