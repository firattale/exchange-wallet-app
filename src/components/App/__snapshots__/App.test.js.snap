// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly in "debug" mode 1`] = `
<div
  className="h-100 w-100"
>
  <Header
    currencyRate=""
    firstAmount=""
    firstCurrency="GBP"
    secondCurrency="USD"
  />
  <div
    className="wallets-container"
  >
    <FirstWallet
      currency="GBP"
      currencyRate=""
      firstAmount=""
      secondCurrency="USD"
      sign="£"
    />
    <SecondWallet
      currency="USD"
      currencyRate=""
      firstCurrency="GBP"
      firstSign="£"
      secondSign="$"
    />
  </div>
</div>
`;
